### PR TITLE
Add tests for shared_app_data.#.dart API examples.

### PR DESCRIPTION
**Pull request by ksokolovskyi**
This PR contributes to https://github.com/flutter/flutter/issues/130459

### Description
- Updates `examples/api/lib/widgets/shared_app_data/shared_app_data.0.dart` to meet latest API examples structure
- Updates `examples/api/lib/widgets/shared_app_data/shared_app_data.1.dart` to meet latest API examples structure
- Adds tests for `examples/api/lib/widgets/shared_app_data/shared_app_data.0.dart`
- Adds tests for `examples/api/lib/widgets/shared_app_data/shared_app_data.1.dart`

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [X] All existing and new tests are passing.

### CODE REVIEW

File: examples/api/lib/widgets/shared_app_data/shared_app_data.0.dart [hunk 1]
```diff
@@ -50,19 +53,27 @@ class _HomeState extends State<Home> {
...
               onPressed: () {
                 _barVersion += 1;
+                // Changing the SharedAppData's value for 'bar' key causes the
+                // widgets that depend on 'bar' key to be rebuilt.
                 SharedAppData.setValue<String, String?>(
-                    context, 'bar', 'BAR $_barVersion'); // no need to call setState()
+                  context,
+                  'bar',
+                  'BAR $_barVersion',
+                ); // no need to call setState()
```
Comment (bleroux) [inline examples/api/lib/widgets/shared_app_data/shared_app_data.0.dart:76]: Very minor: consider capitalizing the first letter and adding a final dot.
Same on line 62.

**Review by bleroux — APPROVED** [review]
LGTM! Thanks for this contribution. Taha and I are attending Google I/O this week so we will probably catch up on pending reviews next week.

File: examples/api/lib/widgets/shared_app_data/shared_app_data.0.dart (hunk 1 above)
Comment (ksokolovskyi) [reply examples/api/lib/widgets/shared_app_data/shared_app_data.0.dart]: @bleroux thanks for the review and the information. Have a great time!

File: examples/api/test/widgets/shared_app_data/shared_app_data.0_test.dart [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+    while (counter < 10) {
+      counter++;
+
+      await tester.tap(
+        find.ancestor(
+          of: find.text('change foo'),
+          matching: find.byType(ElevatedButton),
+        ),
+      );
```
Comment (TahaTesser) [inline examples/api/test/widgets/shared_app_data/shared_app_data.0_test.dart:38]:
```suggestion
      await tester.tap(find.ancestor(
        of: find.text('change foo'),
        matching: find.byType(ElevatedButton),
      ));
```

File: examples/api/test/widgets/shared_app_data/shared_app_data.0_test.dart [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+    while (counter < 10) {
+      counter++;
+
+      await tester.tap(
+        find.ancestor(
+          of: find.text('change bar'),
+          matching: find.byType(ElevatedButton),
+        ),
+      );
```
Comment (TahaTesser) [inline examples/api/test/widgets/shared_app_data/shared_app_data.0_test.dart:60]:
```suggestion
      await tester.tap(find.ancestor(
        of: find.text('change bar'),
        matching: find.byType(ElevatedButton),
      ));
```

**Review by TahaTesser — CHANGES_REQUESTED** [review]
LGTM with nits

**Review by TahaTesser — APPROVED** [review]

**Comment by ksokolovskyi** [comment]
@bleroux @TahaTesser thanks for the review and merge!

**Pull request merged**